interface/patient_file/history/encounters.php

```diff
@@ -379,24 +392,33 @@ function efmouseover(elem, ptid, encid, formname, formid) {
 // $count = 0;
 
 $sqlBindArray = array();
+if($attendant_type == 'pid') {
+    $from = "FROM form_encounter AS fe " .
+        "JOIN forms AS f ON f.pid = fe.pid AND f.encounter = fe.encounter AND " .
+        "f.formdir = 'newpatient' AND f.deleted = 0 ";
+} else {
+    $from = "FROM form_groups_encounter AS fe " .
+        "JOIN forms AS f ON f.therapy_group_id = fe.group_id AND f.encounter = fe.encounter AND " .
+        "f.formdir = 'newGroupEncounter' AND f.deleted = 0 ";
+}
 
-$from = "FROM form_encounter AS fe " .
-  "JOIN forms AS f ON f.pid = fe.pid AND f.encounter = fe.encounter AND " .
-  "f.formdir = 'newpatient' AND f.deleted = 0 ";
 if ($issue) {
   $from .= "JOIN issue_encounter AS ie ON ie.pid = ? AND " .
     "ie.list_id = ? AND ie.encounter = fe.encounter ";
   array_push($sqlBindArray, $pid, $issue);
 }
-$from .= "LEFT JOIN users AS u ON u.id = fe.provider_id WHERE fe.pid = ? ";
-$sqlBindArray[] = $pid;
+if($attendant_type == 'pid') {
+    $from .= "LEFT JOIN users AS u ON u.id = fe.provider_id WHERE fe.pid = ? ";
+} else {
+    $from .= "LEFT JOIN users AS u ON u.id = fe.provider_id WHERE fe.group_id = ? ";
+}
+$sqlBindArray[] = $attendant_type == 'pid' ? $pid : $_SESSION['therapy_group'];
```

Comment:
set the sqlBindArray[] entry in above if/else to improve readability
(note you have already done the if $attendant_type == 'pid' in the above if/else so no reason to do it again)